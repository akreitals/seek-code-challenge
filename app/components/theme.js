const maxContainerWidth = '940px';

const emoji = '"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol"';
const font = `Averta, -apple-system, BlinkMacSystemFont, “Segoe UI”, Roboto, Helvetica, Arial, sans-serif, ${emoji}`;
const mono = 'SFMono-Regular,"Roboto Mono",Menlo,monospace';

const fontSizes = [12, 14, 16, 18, 24, 32, 48, 64, 72, 96];
const space = [0, 4, 8, 16, 32, 64, 128];
const regular = 400;
const bold = 600;

const greys = {
    black: '#1c1c1c',
    charcoal: '#404040',
    midGrayDark: '#747474',
    midGrayMedium: '#898989',
    midGray: '#a8a8a8',
    midGrayLight: '#d6d6d6',
    grayMedium: '#e1e1e1',
    grayLight: '#eeeeee',
    grayLightest: '#f7f7f7',
    offWhite: '#fbfbfb',
    white: '#ffffff'
};

const brand = {
    blueDarker: '#071d40',
    blueDark: '#042763',
    blue: '#0d3880',
    blueLight: '#184da8',
    blueLighter: '#2765cf',
    pink: '#e60278',
    green: '#157e00',
    greenLight: '#5bc252',
    purple: '#9556b7',
    teal: '#00a5ad',
    orange: '#f57c00',
    yellow: '#ffc600',
    yellowLight: '#fffce3',

    primary: '#e60278',
    accent: '#9556b7',
    success: '#157e00',
    info: '#2765cf',
    warning: '#f57c00',
    error: '#e60278'
};

export const colors = {
    ...greys,
    ...brand
};

export const radius = '2px';

// boxShadows
export const boxShadows = [
    `0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)`
];

const theme = {
    colors,
    font,
    mono,
    fontSizes,
    space,
    regular,
    bold,
    maxContainerWidth,
    radius,
    boxShadows
};

export default theme;
