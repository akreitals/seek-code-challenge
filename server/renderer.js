import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
