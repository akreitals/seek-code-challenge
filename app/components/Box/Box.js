import styled from 'styled-components';
import { space, width, color, fontSize, textAlign, borders } from 'styled-system';
import theme from '../theme';

const Box = styled('div')(space, width, color, fontSize, textAlign);

Box.displayName = 'Box';

Box.header = Box.withComponent('header');
Box.main = Box.withComponent('main');
Box.article = Box.withComponent('article');
Box.section = Box.withComponent('section');
Box.footer = Box.withComponent('footer');

Box.propTypes = {
    ...space.propTypes,
    ...width.propTypes,
    ...color.propTypes,
    ...fontSize.propTypes,
    ...textAlign.propTypes,
    ...borders.propTypes
};

Box.defaultProps = {
    theme
};

export default Box;
