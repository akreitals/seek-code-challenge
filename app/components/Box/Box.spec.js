import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import theme, { colors } from '../theme';
import Box from './Box';

describe('Box', () => {
    test('should render a <Box /> component', () => {
        const json = renderer.create(<Box />).toJSON();
        expect(json).toMatchSnapshot();
    });

    test('width prop successfully sets width', () => {
        const json = renderer.create(<Box width={1 / 2} />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('width', '50%');
    });

    test('m prop successfully sets margin', () => {
        const json = renderer.create(<Box m={1} />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('margin', `${theme.space[1]}px`);
    });

    test('p prop successfully sets padding', () => {
        const json = renderer.create(<Box p={1} />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('padding', `${theme.space[1]}px`);
    });

    test('color prop successfully sets color', () => {
        const json = renderer.create(<Box color="warning" />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('color', colors.warning);
    });

    test('bg prop successfully sets background color', () => {
        const json = renderer.create(<Box bg="accent" />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('background-color', colors.accent);
    });

    test('bg prop successfully sets custom background color', () => {
        const json = renderer.create(<Box bg="#ff6d00" />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('background-color', '#ff6d00');
    });
});
