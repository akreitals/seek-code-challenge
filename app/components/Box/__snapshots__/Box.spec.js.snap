// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box bg prop successfully sets background color 1`] = `
.c0 {
  background-color: #9556b7;
}

<div
  className="c0"
/>
`;

exports[`Box bg prop successfully sets custom background color 1`] = `
.c0 {
  background-color: #ff6d00;
}

<div
  className="c0"
/>
`;

exports[`Box color prop successfully sets color 1`] = `
.c0 {
  color: #f57c00;
}

<div
  className="c0"
  color="warning"
/>
`;

exports[`Box m prop successfully sets margin 1`] = `
.c0 {
  margin: 4px;
}

<div
  className="c0"
/>
`;

exports[`Box p prop successfully sets padding 1`] = `
.c0 {
  padding: 4px;
}

<div
  className="c0"
/>
`;

exports[`Box should render a <Box /> component 1`] = `
<div
  className=""
/>
`;

exports[`Box width prop successfully sets width 1`] = `
.c0 {
  width: 50%;
}

<div
  className="c0"
  width={0.5}
/>
`;
